packages/grafana-alerting/scripts/codegen.ts

```diff
@@ -1,21 +1,48 @@
 /**
- * This script will generate TypeScript type definitions and a RTKQ client for the alerting k8s APIs.
- * It downloads the OpenAPI schema from a running Grafana instance and generates the types.
+ * This script will generate TypeScript type definitions and a RTKQ clients for the alerting k8s APIs.
  *
  * Run `yarn run codegen` from the "grafana-alerting" package to invoke this script.
+ *
+ * API clients will be placed in "src/grafana/api/<version>/api.gen.ts"
  */
-import { type ConfigFile } from '@rtk-query/codegen-openapi';
+import type { ConfigFile } from '@rtk-query/codegen-openapi';
 import { resolve } from 'node:path';
 
-// these snapshots are generated by running "go test pkg/tests/apis/openapi_test.go", see the README in the "openapi_snapshots" directory
-const OPENAPI_SCHEMA_LOCATION = resolve(
-  '../../../pkg/tests/apis/openapi_snapshots/notifications.alerting.grafana.app-v0alpha1.json'
-);
+// ℹ️ append versions here to generate additional API clients
+const VERSIONS = ['v0alpha1'] as const;
+
+type OutputFile = Omit<ConfigFile, 'outputFile'>;
+type OutputFiles = Record<string, OutputFile>;
+
+const outputFiles = VERSIONS.reduce<OutputFiles>((acc, version) => {
+  // we append the version here so we export versioned API clients from this package without having to re-export with an alias
+  const exportName = `alertingAPI${version}`;
```

Comment:
I'm not sure you need to have the version in the API name if you're scoping version per folder.

Reply:
Good point, I was doing this to avoid having to export with an alias in `unstable.ts` but thinking about this a bit more that's probably the more sensible thing to do 🤔